kong/conf_loader.lua

```diff
@@ -425,60 +421,35 @@ local function check_and_infer(conf, opts)
           found = true
+          break
         end
       end
 
-      if not found then
-        if is_cassandra then
-          errors[#errors + 1] = fmt("%s has an invalid value: '%s' (%s)", name,
-                                    option[1],
-                                    concat(CASSANDRA_CONSISTENCY, ", "))
-        end
-
-      else
+      if found then
         read_consistency  = option[1]
-        write_consistency = option[1]
-      end
-    end
+        write_consistency = read_consistency
 
-    if not read_consistency then
-      read_consistency  = conf.cassandra_consistency
-    end
-
-    if not write_consistency then
-      write_consistency = conf.cassandra_consistency
-    end
-
-    if read_consistency and write_consistency then
-      if opts.starting and is_cassandra and is_dc_aware then
-        local suggestion
-        if read_consistency == "QUORUM" or
-           read_consistency  == "ONE" then
-          if write_consistency == "QUORUM" or
-             write_consistency == "ONE" then
-            suggestion = fmt("read:LOCAL_%s, write:LOCAL_%s",
-                             read_consistency, write_consistency)
-          else
-            suggestion = fmt("read:LOCAL_%s, write:%s",
-                             read_consistency, write_consistency)
-          end
-        elseif write_consistency == "QUORUM" or
-               write_consistency == "ONE" then
-          suggestion = fmt("read:%s, write:LOCAL_%s",
-                           read_consistency, write_consistency)
-        end
+        option.read  = read_consistency:lower()
+        option.write = write_consistency:lower()
 
-        if suggestion then
-          log.warn("Cassandra is configured to use a data center aware "  ..
-                   "load balancing strategy '%s' while the consistency "  ..
-                   "of '%s' is set to '%s'. It is recommended to change " ..
-                   "it to '%s'.", conf.cassandra_lb_policy, name,
-                   concat(option, ", "), suggestion)
-        end
+      else
+        errors[#errors + 1] = fmt("%s has an invalid value: '%s' (%s)", name,
+                                  option[1],
+                                  concat(CASSANDRA_CONSISTENCY, ", "))
       end
+    end
 
-      option.read  = read_consistency:lower()
-      option.write = write_consistency:lower()
+    if is_cassandra and is_dc_aware and ((
+      read_consistency ~= "LOCAL_QUORUM"  and read_consistency ~= "LOCAL_ONE"
+    ) or (
+      write_consistency ~= "LOCAL_QUORUM" and write_consistency ~= "LOCAL_ONE"
+    )) then
+      log.warn("Cassandra is configured to use a data center aware load "    ..
+               "balancing strategy '%s' while the consistency of '%s' is "   ..
+               "set to '%s'. It is recommended to use 'LOCAL_QUORUM' or "    ..
+               "'LOCAL_ONE' when specifying read/write consistency in such " ..
+               "scenario.", conf.cassandra_lb_policy, name,
```

Comment:
Should be `scenarios` (or "scenarii" if we wanted to be pedantic about it, but let's avoid it) .

Or maybe we could just simplify this warning:

```
A datacenter-aware Cassandra load balancing policy is in use ('cassandra_lb_policy=%s')
but some consistency levels are not datacenter-aware (e.g. LOCAL_QUORUM, LOCAL_ONE, or EACH_QUORUM).
```

And we only print the warning once, even if both `cassandra_consistency_admin` and `cassandra_consistency_proxy` are mis-configured.

Reply:
(Off-topic, but since I was linguistics-nerd-sniped into it: [scenarii](https://english.stackexchange.com/questions/11775/what-is-the-plural-of-scenario) is a case of [hypercorrection](https://en.wikipedia.org/wiki/Hypercorrection) — "scenarios" is in fact the formally correct form.)